File: api/getpost.js
const express = require('express');
const mysql = require('mysql2/promise');

const app = express();
const port = 3000;

// Configuração do banco
const dbConfig = {
  host: 'mysql-container',
  user: 'root',
  password: 'senha123',
  database: 'meuapp'
};


app.use(express.json());


app.get('/', async (req, res) => {
  try {
    const connection = await mysql.createConnection(dbConfig);
    const [rows] = await connection.execute('SELECT id FROM status');
    await connection.end();
    
    const quantidade = rows[0].id;
    res.json({ registros: quantidade });
  } catch (error) {
    res.status(500).json({ error: error.message });
  }
});


app.get('/:id', async (req, res) => {
  try {
    const connection = await mysql.createConnection(dbConfig);
    const [rows] = await connection.execute('SELECT valor FROM eventos WHERE id = ?', [req.params.id]);
    await connection.end();
    
    if (rows.length === 0) {
      res.status(404).json({ error: 'Registro não encontrado' });
    } else {
      res.json({ id: req.params.id, valor: rows[0].valor });
    }
  } catch (error) {
    res.status(500).json({ error: error.message });
  }
});

app.post('/incrementar', async (req, res) => {
  try {
    const connection = await mysql.createConnection(dbConfig);
    await connection.execute('UPDATE status SET id = id + 1');
    const [rows] = await connection.execute('SELECT id FROM status');
    await connection.end();
    
    res.json({ novo_id: rows[0].id });
  } catch (error) {
    res.status(500).json({ error: error.message });
  }
});

// Rota POST /incrementar-async - incrementa o status (sem retorno imediato)
app.post('/incrementar-async', async (req, res) => {
  try {
    const connection = await mysql.createConnection(dbConfig);
    await connection.execute('UPDATE status SET id = id + 1');
    await connection.end();
    

    res.json({ message: 'Incremento realizado' });
  } catch (error) {
    res.status(500).json({ error: error.message });
  }
});

app.listen(port, '0.0.0.0', () => {
  console.log('App rodando na porta ' + port);
});

File: api/getpost.py
from flask import Flask, jsonify, request
import mysql.connector

app = Flask(__name__)
port = 3000


db_config = {
    'host': 'mysql-container',  
    'user': 'root',
    'password': 'senha123',
    'database': 'meuapp',
}

# Função para abrir conexão
def get_connection():
    return mysql.connector.connect(**db_config)


@app.route('/', methods=['GET'])
def get_registros():
    try:
        conn = get_connection()
        cursor = conn.cursor(dictionary=True)
        cursor.execute("SELECT id FROM status")
        rows = cursor.fetchall()
        cursor.close()
        conn.close()

        if rows:
            quantidade = rows[0]['id']
        else:
            quantidade = 0

        return jsonify({'registros': quantidade})
    except Exception as e:
        return jsonify({'error': str(e)}), 500

@app.route('/<int:id>', methods=['GET'])
def get_evento(id):
    try:
        conn = get_connection()
        cursor = conn.cursor(dictionary=True)
        cursor.execute("SELECT valor FROM eventos WHERE id = %s", (id,))
        row = cursor.fetchone()
        cursor.close()
        conn.close()

        if row:
            return jsonify({'id': id, 'valor': row['valor']})
        else:
            return jsonify({'error': 'Registro não encontrado'}), 404
    except Exception as e:
        return jsonify({'error': str(e)}), 500


@app.route('/incrementar', methods=['POST'])
def incrementar():
    try:
        conn = get_connection()
        cursor = conn.cursor()
        cursor.execute("UPDATE status SET id = id + 1")
        conn.commit()
        cursor.execute("SELECT id FROM status")
        row = cursor.fetchone()
        cursor.close()
        conn.close()

        if row: 
            return jsonify({'novo_id': row[0]})
        else:
            return jsonify({'error': 'Nenhum registro encontrado'}), 404
            
    except Exception as e:
        return jsonify({'error': str(e)}), 500
    
@app.route('/incrementar-async', methods=['POST'])
def incrementar_async():
    try:
        conn = get_connection()
        cursor = conn.cursor()
        cursor.execute("UPDATE status SET id = id + 1")
        conn.commit()
        cursor.close()
        conn.close()

        return jsonify({'message': 'Incremento realizado'})
    except Exception as e:
        return jsonify({'error': str(e)}), 500

if __name__ == '__main__':
    app.run(host='0.0.0.0', port=port)
